From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Thu, 30 Jul 2020 23:53:11 +0000
Subject: Revert "ANDROID: ASoC: compress: fix unsigned integer overflow check"

Revert submission 144587-1572027951673-b7d38b4

Reason for revert: See b/142489397
Reverted Changes:
Ia6d475540:ANDROID: ASoC: compress: fix unsigned integer over...
If77f8b37b:ANDROID: ALSA: jack: Update supported jack switch ...

Change-Id: I514458a88b3b96bc2da6d1558b41fd3ccf0aafbd
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 sound/core/compress_offload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/core/compress_offload.c b/sound/core/compress_offload.c
--- a/sound/core/compress_offload.c
+++ b/sound/core/compress_offload.c
@@ -570,7 +570,7 @@ static int snd_compress_check_input(struct snd_compr_params *params)
 {
 	/* first let's check the buffer parameter's */
 	if (params->buffer.fragment_size == 0 ||
-	    params->buffer.fragments > U32_MAX / params->buffer.fragment_size ||
+	    params->buffer.fragments > INT_MAX / params->buffer.fragment_size ||
 	    params->buffer.fragments == 0)
 		return -EINVAL;
 
